File: 746_Min_Cost_Climbing_Stairs.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        f1 = cost[0] 
        f2 = cost[1]
        for i in range(2,len(cost)):
            f1, f2 = f2,cost[i] + min(f1, f2)
        return min(f1, f2)

File: 746_Min_Cost_Climbing_Stairs.cpp
   int minCostClimbingStairs(vector<int>& c) {
        int n=c.size();
        int dp[n+2];
        
        dp[0]=c[0];
        dp[1]=c[1];
        
        for(int i=2;i<n;i++){
            dp[i]=min(c[i]+dp[i-1],c[i]+dp[i-2]);
        }
        
        dp[n]=min(dp[n-1],dp[n-2]);
        
        return dp[n];
    }